src/markdown.cpp

```diff
@@ -3107,30 +3129,65 @@ void MarkdownOutlineParser::parseInput(const QCString &fileName,
   QCString mdfileAsMainPage = Config_getString(USE_MDFILE_AS_MAINPAGE);
   bool wasEmpty = id.isEmpty();
   if (wasEmpty) id = markdownFileNameToId(fileName);
-  if (!isExplicitPage(docs))
-  {
-    if (!mdfileAsMainPage.isEmpty() &&
-        (fn==mdfileAsMainPage || // name reference
-         FileInfo(fileName.str()).absFilePath()==
-         FileInfo(mdfileAsMainPage.str()).absFilePath()) // file reference with path
-       )
-    {
-      docs.prepend("@anchor " + id + "\\ilinebr ");
-      docs.prepend("@mainpage "+title+"\\ilinebr ");
-    }
-    else if (id=="mainpage" || id=="index")
-    {
-      if (title.isEmpty()) title = titleFn;
-      docs.prepend("@anchor " + id + "\\ilinebr ");
-      docs.prepend("@mainpage "+title+"\\ilinebr ");
-    }
-    else
-    {
-      if (title.isEmpty()) {title = titleFn;prepend=0;}
-      if (!wasEmpty) docs.prepend("@anchor " +  markdownFileNameToId(fileName) + "\\ilinebr ");
-      docs.prepend("@page "+id+" "+title+"\\ilinebr ");
-    }
-    for (int i = 0; i < prepend; i++) docs.prepend("\n");
+  static const reg::Ex re(R"([\\@]page[ \t]*)");
+  static const reg::Ex re0(R"(^[a-z_A-Z\x80-\xFF])"); // from LABELID, see commentscan.l
+  static const reg::Ex re1(R"([a-z_A-Z\x80-\xFF][a-z_A-Z0-9\x80-\xFF\-]*)"); // LABELID, see commentscan.l
+  static const reg::Ex re2(R"(\n)");
+  reg::Match match;
+  reg::Match match1;
+  std::string t;
+  std::string t1;
+  switch (isExplicitPage(docs))
```

Comment:
you may want to move the variable definitions to the innermost scope where they are used (in this case, the body of the switch... case explicitPage branch could be put into a block scope {} ). see also:  https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rnr-top

Reply:
Thanks for the feedback.
Indeed, I had the code there before and got warnings like:
```
...\doxygen\src\markdown.cpp(3191): error C2360: initialization of 't1' is skipped by 'case' label
...\doxygen\src\markdown.cpp(3167): note: see declaration of 't1'
```

but that was because I forgot to put the case part here in a block scope.